UPD: Update conda and pip packages for Python 3.11

vhosouza opened the pull request:
Update several relevant package versions in both conda and environment.

This solves issues with installation in MacOS due to old version of rtmidi.

It also bumps vtk, nibabel, pyacvd, and several other packages to the latest version.

I tried to test the new numpy 2.0, but raised errors on cython compilation due to deprecated IF in pyxd file. So, I leaving this update to a later stage.

I tested in MacOS Sonoma with conda and pip installations and image loading, tractography, debug navigation are working fine.

Known issue:

I could not create a surface after brain segmentation which raised the following error:

![image](https://github.com/invesalius/invesalius3/assets/1081406/ab0e5bd5-d0e4-48b8-801a-a2d1aa261cbf)

@tfmoraes can you test it, please?

tfmoraes commented:
About numpy 2.0 is better to wait. Some libraries are having problem with numpy 2.0.

tfmoraes commented on requirements.txt:
```diff
@@ -1,25 +1,25 @@
 # Scientific
 numpy==1.24.3
```
I think it's good idea to update numpy to last version before 2.0 (1.26.4)

tfmoraes commented on environment.yml:
```diff
@@ -2,37 +2,40 @@ channels:
+  - h5py==3.11.0
+  - imageio==2.34.2
+  - nibabel==5.2.1
   - numpy==1.24.3
```
I think it's good idea to update numpy to last version before 2.0 (1.26.4)

tfmoraes commented:
Hi @vhosouza what is the uname of this macos? 

```
$ uname -a
```

vhosouza commented:
> uname -a

Darwin ****** 23.5.0 Darwin Kernel Version 23.5.0: Wed May  1 20:14:38 PDT 2024; root:xnu-10063.121.3~5/RELEASE_ARM64_T6020 arm64

vhosouza commented:
> Hi @vhosouza what is the uname of this macos?
> 
> ```
> $ uname -a
> ```

@tfmoraes, I think the problem with surface creation was because I updated the packages but the "setup.py" was not updated. I did run the setup.py, but because the files already existed, I think it did not overwrite them, so they were still based on the old environment. I deleted the cythonized files and compiled them again, now everything is working fine, including the brain segmentation and surface creation.

If you don't have any further issues, this PR is ready to merge. @rmatsuda and @tfmoraes

tfmoraes commented:
@vhosouza good to know that. I tested here with a Macbook Air M1 with last MacOS version and surface creation (using CA Smoothing) was running without any problems. I was even working in a C++ version of CA Smoothing (https://github.com/tfmoraes/invesalius3/tree/mesh_h) to fix this problem.

vhosouza commented:
Thanks @tfmoraes!! If all is fine, can you accept this PR?
